decimalSuite_test.go

```diff
@@ -0,0 +1,138 @@
+	dataString := string(dat)
+	//  "(\n|\r)") // start with newline
+	//  "(?P<TestName>dd[\w]*)") // TestName must start with dd (double decimal)
+	//  "(?P<TestFunc>[\S]*)") // testfunc made of anything that isn't a whitespace
+	//  "(\s*\'?)") // after can be any number of spaces and quotations if they exist
+	// "(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	//  "(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	// ('?\s*'?) match any quotation marks and any spaces
+	// (?P<TestVals_2>(\+|-[^->])?[^\r\n\t\f\v\' ]*) testvals2 same as 1 but no '->'
+	// ('?\s*->\s*'?) matches the indicator to answer
+	// (?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*) matches the answer that's anything that is plus minus but not quotations
+
+	// split into groups: testName, TestFunct, TestVals (x2) and TestAns
+	regex := `(\n|\r)(?P<TestName>dd[\w]*)(\s*)(?P<TestFunc>[\S]*)(\s*\'?)(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)('?\s*'?)(?P<TestVals_2>(\+|-[^->])?[^\r\n\t\f\v\' ]*)('?\s*->\s*'?)(?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*)`
+	r := regexp.MustCompile(regex)
+	ans := r.FindAllStringSubmatch(dataString, -1)
+	testList := []string{}
+	var data = map[string]map[string]string{}
+
+	for i := 0; i < len(ans); i++ {
+		data[ans[i][2]] = make(map[string]string)
+		data[ans[i][2]]["TestFunc"] = ans[i][4]
+		data[ans[i][2]]["TestVal1"] = ans[i][6]
+		data[ans[i][2]]["TestVal2"] = ans[i][9]
+		data[ans[i][2]]["TestAns"] = ans[i][12]
+		testList = append(testList, ans[i][2])
+
+	}
+	return data, testList
+
+}
+
+//convertToDec64 converts the map object strings to decimal64s
+func convertToDec64(testVals map[string]string, testName string) (testVal1, testVal2, testAns Decimal64, parseErr string) {
+	var err1, err2, err3 error
+	// var TestResult []string
+
+	testVal1, err1 = ParseDecimal64(testVals["TestVal1"])
+	testAns, err3 = ParseDecimal64(testVals["TestAns"])
+
+	testVal2, err2 = ParseDecimal64(testVals["TestVal2"])
+	parseErr = checkErrors(err1, err2, err3, testVals)
+
+	return
+}
+
+//checkErrors checks to see if there are any string parsing errors
+func checkErrors(err1, err2, err3 error, testVals map[string]string) (errReturn string) {
+	var errReturnSlice []string
+	if err2 != nil {
+		errReturnSlice = append(errReturnSlice, fmt.Sprintf("error parsing input of value: %s \n", testVals["TestVal2"]))
+	}
+	if err1 != nil {
+		errReturnSlice = append(errReturnSlice, fmt.Sprintf("error parsing input of value: %s \n", testVals["TestVal1"]))
+	}
+
+	if err3 != nil {
+		errReturnSlice = append(errReturnSlice, fmt.Sprintf("error parsing input of value: %s \n", testVals["TestAns"]))
+	}
+	return strings.Join(errReturnSlice, "")
+}
+
+//doTests completes the tests and returns a boolean and string on if the test passes
+func doTests(testVal1, testVal2, testAns Decimal64, testfunc string) (testStatus bool, testString string) {
```

Comment:
I'd rather let it just return a bool of if the test passed, 
I want to keep it this way because I need to figure out the logic behind Nan equalities that will be a little bit complicated if I do it within the calling function

Reply:
sure